src/main/java/net/javadiscord/javabot/systems/moderation/AutoMod.java

```diff
@@ -202,11 +202,10 @@ public boolean hasSuspiciousLink(Message message) {
 				String url = urlMatcher.group(0).trim();
 				try {
 					URI uri = new URI(url);
-					if (spamUrls.contains(uri.getHost())) {
+					if (uri.getHost() != null && spamUrls.contains(uri.getHost())) {
 						return true;
 					}
-				} catch (URISyntaxException e) {
-					e.printStackTrace();
+				} catch (URISyntaxException ignored) {
```

Comment:
Why did you just ignore the exception without even logging it?

Reply:
How does removing the printing of the stack trace stop the problem from occurring in the first place?

Reply:
As long as it works correctly

Reply:
Note that `As long as it works correctly` is no excuse for bad code in any way.

Reply:
At least put something in the log indicating that this error occurred, swallowing exceptions completely like this is not something you should do